lib/note-detail/toggle-task/constants.js

```diff
@@ -3,4 +3,4 @@ export const selectors = {
   markdownRoot: '[data-markdown-root]',
 };
 
-export const taskRegex = /^(\s+)?-[ \t]+\[[xX\s]?\](.+)/gm;
+export const taskRegex = /^(\s+)?(-[ \t]+\[[xX\s]?\])(.+)/gm;
```

Comment:
I don't think we want the whitespace matching in this regex, I'm able to add spaces and then add a checkbox:

![spaces-checkbox](https://user-images.githubusercontent.com/789137/51229547-a7b05380-1911-11e9-9d16-a4d31031e432.gif)

Reply:
This is because Markdown lists can be nested:

```
- [ ] parent item
  - [ ] sub item
```

- [ ] parent item
  - [ ] sub item

Reply:
Oh, ok. The mobile apps enforce them to be at the start of the line. Should we do that here for consistency?

Reply:
Functionally, I don't think it's unreasonable to enforce that tasklists be flat. Rather, I'm more concerned about straying too much from the GFM spec (or common implementations), as it will hurt our compatibility within the wider Markdown ecosystem. Users do import/export and copy/paste between Markdown editors. Not to mention internal consistency, with the Markdown renderers we're using for previews.

Disregarding code blocks (when rendering checkboxes) was a trade-off we had to make for feasibility reasons, but that’s not an issue here with the leading whitespace. I'm leaning toward consistency with common Markdown implementations, as long as it's feasible. Are there any extra considerations on the mobile side?

Reply:
Let me investigate how hard it'd be to support it on the mobile apps.

Reply:
OK it should be possible 👍I'll create PRs for the other apps to support the nested checklists.

Reply:
Awesome. Thank you!